packages/theming/src/utils/arrowStyles.ts

```diff
@@ -0,0 +1,188 @@
+      animation: 0.3s ease-in-out ${animationName};
+    }
+  `;
+};
+
+const positionStyles = (position: ARROW_POSITION, size: string, inset: string) => {
+  const margin = math(`${size} / -2`);
+  const placement = math(`${margin} - ${inset}`);
+  let clipPath;
+  let positionCss;
+  let propertyRadius: string;
+
+  if (position.startsWith('top')) {
+    propertyRadius = 'border-bottom-right-radius';
+    clipPath = 'polygon(100% 0, 100% 1px, 1px 100%, 0 100%, 0 0)';
+    positionCss = css`
+      top: ${placement};
+      right: ${position === 'top-right' && size};
+      left: ${position === 'top' ? '50%' : position === 'top-left' && size};
+      margin-left: ${position === 'top' && margin};
+    `;
+  } else if (position.startsWith('right')) {
+    propertyRadius = 'border-bottom-left-radius';
+    clipPath = 'polygon(100% 0, 100% 100%, calc(100% - 1px) 100%, 0 1px, 0 0)';
+    positionCss = css`
+      top: ${position === 'right' ? '50%' : position === 'right-top' && size};
+      right: ${placement};
+      bottom: ${position === 'right-bottom' && size};
+      margin-top: ${position === 'right' && margin};
+    `;
+  } else if (position.startsWith('bottom')) {
+    propertyRadius = 'border-top-left-radius';
+    clipPath = 'polygon(100% 0, calc(100% - 1px) 0, 0 calc(100% - 1px), 0 100%, 100% 100%)';
+    positionCss = css`
+      right: ${position === 'bottom-right' && size};
+      bottom: ${placement};
+      left: ${position === 'bottom' ? '50%' : position === 'bottom-left' && size};
+      margin-left: ${position === 'bottom' && margin};
+    `;
+  } else if (position.startsWith('left')) {
+    propertyRadius = 'border-top-right-radius';
+    clipPath = 'polygon(0 100%, 100% 100%, 100% calc(100% - 1px), 1px 0, 0 0)';
+    positionCss = css`
+      top: ${position === 'left' ? '50%' : position === 'left-top' && size};
+      bottom: ${size};
+      left: ${placement};
+      margin-top: ${position === 'left' && margin};
+    `;
+  }
+
+  /**
+   * 1. Round-off portion of the foreground square opposite the arrow tip
+   *    (improved layout for IE which doesn't support 'clip-path').
+   * 2. Clip portion of the foreground square opposite the arrow tip so that it
+   *    doesn't interfere with container content.
+   * 3. Arrow positioning on the base element.
+   */
+  return css`
+    &::before {
+      ${propertyRadius!}: 100%; /* [1] */
+      clip-path: ${clipPath}; /* [2] */
+    }
+
+    &::before,
+    &::after {
+      ${positionCss} /* [3] */
+    }
+  `;
+};
+
+/**
+ * CSS for an arrow at the given position and with the given size. The arrow is
+ * positioned via `::before` and `::after` pseudo-elements and inherits the
+ * base element's border, background, and box-shadow.
+ *
+ * IMPORTANT: the element that receives arrow styling must be wrapped in a
+ * positioned box (i.e. `relative`, `absolute`, `fixed`) that has a `z-index`
+ * greater than or equal to zero. Without this the arrow cannot properly
```

Comment:
nit (comma): "Without this, the arrow cannot properly display inherited border or box-shadow styling."